### PR TITLE
Extract applyEntry from apply

### PR DESCRIPTION
**Pull request by serathius**
cc @ahrtr @fuweid @chaochn47 @siyuanfoundation

### CODE REVIEW

File: server/etcdserver/server.go [hunk 1]
```diff
@@ -1771,50 +1771,56 @@ func (s *EtcdServer) apply(
...
-			)
-		}
+		removedSelf := s.applyEntry(&e, confState, raftAdvancedC)
+		shouldStop = shouldStop || removedSelf
 		appliedi, appliedt = e.Index, e.Term
 	}
 	return appliedt, appliedi, shouldStop
 }
 
+func (s *EtcdServer) applyEntry(e *raftpb.Entry, confState *raftpb.ConfState, raftAdvancedC <-chan struct{}) (shouldStop bool) {
```
Comment (fuweid) [inline server/etcdserver/server.go:1781]: IIUC, `shouldStop` should be `removedSelf`

**Review by fuweid — APPROVED** [review]
LGTM

File: server/etcdserver/server.go (hunk 1 above)
Comment (serathius) [reply server/etcdserver/server.go:1781]: done

**Comment by ahrtr** [comment]
I do not see much value to simplify an already very simple method `apply` (only 50 lines of code, and the structure is also very clear). 

It also increases the depth of call stack.

**Comment by ahrtr** [comment]
It is also weird that `applyEntry` returns a `removedSelf`

**Comment by serathius** [comment]
Goal of this refactor is to extract applyEntry so we can unify the defer logic applying consistent index between two branches (applyNormalEntry and applyConfChange)

**Comment by fuweid** [comment]
> It also increases the depth of call stack.

The go1.21 starts inline everything 😂

**Pull request closed without merging**